### PR TITLE
jnp.frexp: add custom JVP rule for proper derivatives.

### PR DESCRIPTION
**Pull request by jakevdp**
This is in service of #28040: it's possible we will fix `ldexp` by using `frexp` in its implementation; but that approach fails the `ldexp` gradient tests unless `frexp` can be differentiated.

### CODE REVIEW

File: jax/_src/numpy/ufuncs.py [hunk 1]
```diff
@@ -3061,6 +3064,17 @@ def frexp(x: ArrayLike, /) -> tuple[Array, Array]:
   return _where(cond, x, x1), lax.convert_element_type(x2, np.int32)
 
 
+@_frexp.defjvp
+def _frexp_jvp(primals, tangents):
+  x, = primals
+  t, = tangents
+  m, e = frexp(x)
+  # mdot = ldexp(t, -e)
+  mdot = t * exp2(-e)
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:3073]: We maybe should use `ldexp` here, as it's more accurate for large values of `e`. However that results in failing this assertion:
```
  File "jax/_src/lax/lax.py", line 7310, in _select_transpose_rule
    assert not ad.is_undefined_primal(which)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
```

Comment (jakevdp) [reply jax/_src/numpy/ufuncs.py:3073]: Actually, maybe I shouldn't worry about this: the only problematic cases are when `exp2(-e)` overflows. but by construction, `e` is the exponent of a non-overflowing value, so it should be in range.

**Review by froystig — APPROVED** [review]
